bcda/auth/plugin/alpha.go

```diff
@@ -20,18 +65,86 @@ func (p *AlphaAuthPlugin) DeleteClient(params []byte) error {
 	return errors.New("Not yet implemented")
 }
 
+// can treat as a no-op or call RequestAccessToken
 func (p *AlphaAuthPlugin) GenerateClientCredentials(params []byte) ([]byte, error) {
 	return nil, errors.New("Not yet implemented")
 }
 
+// look up the active access token associated with id, and call RevokeAccessToken
 func (p *AlphaAuthPlugin) RevokeClientCredentials(params []byte) error {
 	return errors.New("Not yet implemented")
 }
 
+// generate a token for the id (which user? just have a single "user" (alpha2, alpha3, ...) per test cycle?)
 func (p *AlphaAuthPlugin) RequestAccessToken(params []byte) (jwt.Token, error) {
-	return jwt.Token{}, errors.New("Not yet implemented")
+	backend := auth.InitAuthBackend()
+	db := database.GetGORMDbConnection()
+	jwtToken := jwt.Token{}
+
+	var (
+		aco models.ACO
+		user models.User
+		err error
+		j interface{}
+	)
+
+	if err = json.Unmarshal(params, &j); err != nil {
+		return jwtToken, err
+	}
+	paramsMap := j.(map[string]interface{})
+
+	acoUUIDString, ok := paramsMap["clientID"].(string)
+	if !ok {
+		return jwtToken, errors.New("missing or otherwise invalid clientID")
+	}
+
+	if aco, err = getFromDB(acoUUIDString); err != nil {
+		return jwtToken, err
+	}
+
+	// I arbitrarily decided to use the first user. An alternative would be to make a specific user
+	// that represents the client. I have no strong opinion here other than not creating stuff in the db
+	// unless we're willing to live with it forever.
+	if err = db.First(&user, "aco_id = ?", aco.UUID).Error; err != nil {
+		return jwtToken, errors.New("No user found for " + aco.UUID.String())
+	}
+
+	ttl, ok := paramsMap["ttl"].(int)
+	if !ok {
+		ttl = 72
+	}
+
+	tokenUUID := uuid.NewRandom()
+	jwtToken = *jwt.New(jwt.SigningMethodRS512)
+	jwtToken.Claims = jwt.MapClaims{
+		"exp": time.Now().Add(time.Hour * time.Duration(ttl)).Unix(),
+		"iat": time.Now().Unix(),
+		"sub": user.UUID.String(),
+		"aco": aco.UUID.String(),
+		"id":  tokenUUID.String(),
+	}
+
+	tokenString, err := backend.SignJwtToken(jwtToken)
+	if err != nil {
+		return jwtToken, err
+	}
+
+	token := auth.Token{
+		UUID:   tokenUUID,
+		UserID: user.UUID,
+		Value:  tokenString,
+		Active: true,
+		Token:  jwtToken,
+	}
+
+	if err = db.Create(&token).Error; err != nil {
+		return jwtToken, err
+	}
+
+	return jwtToken, err // really want to return auth.Token here, but first let's get this all working
```

Comment:
Should we have the return type be `auth.Token` in the interface?

Reply:
yes. I want to do that in the next few changes, where we actually start working with the tokens.